test/performance/perf_test.py

```diff
@@ -429,11 +457,141 @@ def do_task(self):
         self.corpus.insert_many(self.documents, ordered=True)
 
 
+class TestSmallDocClientBulkInsert(SmallDocInsertTest, unittest.TestCase):
+    @client_context.require_version_min(8, 0, 0, -24)
+    def setUp(self):
+        super().setUp()
+        self.models = []
+        for doc in self.documents:
+            self.models.append(InsertOne(namespace="perftest.corpus", document=doc.copy()))
+
+    @client_context.require_version_min(8, 0, 0, -24)
+    def do_task(self):
+        self.client.bulk_write(self.models, ordered=True)
+
+
+class TestSmallDocBulkMixedOps(SmallDocMixedTest, unittest.TestCase):
+    def setUp(self):
+        super().setUp()
+        self.models = []
+        for doc in self.documents:
+            self.models.append(InsertOne(document=doc.copy()))
+            self.models.append(ReplaceOne(filter={}, replacement=doc.copy(), upsert=True))  # type: ignore[arg-type]
+            self.models.append(DeleteMany(filter={}))  # type: ignore[arg-type]
+
+    def do_task(self):
+        self.corpus.bulk_write(self.models, ordered=True)
+
+
+class TestSmallDocClientBulkMixedOps(SmallDocMixedTest, unittest.TestCase):
+    @client_context.require_version_min(8, 0, 0, -24)
+    def setUp(self):
+        super().setUp()
+        self.models = []
+        for doc in self.documents:
+            self.models.append(InsertOne(namespace="perftest.corpus", document=doc.copy()))
+            self.models.append(
+                ReplaceOne(
+                    namespace="perftest.corpus", filter={}, replacement=doc.copy(), upsert=True
+                )  # type: ignore[arg-type]
```

Comment:
What's the type ignore for?

Reply:
Appending the ReplaceOne and DeleteMany to self.models was causing a type error (since it was expecting self.models to be a list of InsertOne's after the first model). I can add a type hint to self.models and remove the type ignores.